osu.Server.Queues.ScoreStatisticsProcessor/Processors/UserTotalPerformanceProcessor.cs

```diff
@@ -85,7 +83,7 @@ public async Task UpdateUserStatsAsync(UserStats userStats, int rulesetId, MySql
 
             (userStats.rank_score, userStats.accuracy_new) = UserTotalPerformanceAggregateHelper.CalculateUserTotalPerformanceAggregates(userStats.user_id, scores);
 
-            if (updateIndex)
+            if (updateMilestones)
```

Comment:
At this point you could argue that instead of having a flag for this you just remove it and have the one usage that passes `updateMilestones: true` call `updateGlobalRank()` itself after calling this method and the other one that passes `false` - just not. Not sure which is better. I'd argue this is already twisty and confusing given `UpdateUserStatsAsync()` is xmldoc'd to "not insert new stats values into the database" but `updateGlobalRank()` *does* write to database, and making that change I propose would kinda fix that.

Reply:
Sounds good, adjusted.